### DISCUSSION TITLE
Supervisors And Workers, How to Use as chatBot?

### DISCUSSION THREAD

**Discussion by ahgsql** [FlowiseAI/Flowise#2820] [Q&A]
So my need, to help my customers in my law firm, I want to create a customer Represantative Supervisor, and two workers. 

1) worker - will answer questions related to law and articles that we publish on our webpage ( have access to tools)
2) worker -  Check all required information gathered to create appoitment. Eg name, phone and appoitment date.

When i start conversation, Supervisor pass task to 1. worker then 2. worker , than after 4-5 iterations, supervisor FINISH. 

What i expect them to in first stage, without assigning 2. worker, they should response me like "How can i help you? You need information about our services or create an appoitment"

Than lets say customer answered with "I want to create an appoitment",

chat response should be : " Please provide your name, phone and date for the appoitment",

than if customer replied with only name and phone, 2. worker should reply with "Required date"

Than chat response should be "Please give date for appoitment". ... ETC

But current workflow is not working as expected.

Attached my workflow, Help me to achieve this kind of system

[Güleç Müşteri Hizmetleri Agents.json](https://github.com/user-attachments/files/16273591/Gulec.Musteri.Hizmetleri.Agents.json)

**Comment by jars08** [comment]
I'm unsure if a multi-agent system is the right thing for that. You can set a starter-message in the Settings for you flow. But maybe for this task a conditional Chatflow that sets you to specialized agents would be better. The multi-agent system is strong in dividing a task into smaller subtasks and refines the output with the different system prompts of the different workers, but you always just need one or the other if I'm understanding you right.
